testplans/graphsync/main.go

```diff
@@ -538,7 +538,7 @@ func runProvider(ctx context.Context, runenv *runtime.RunEnv, initCtx *run.InitC
 				// set up http server to send file
 				http.HandleFunc(fmt.Sprintf("/%s", node.Cid()), func(w http.ResponseWriter, r *http.Request) {
 					fileReader, err := os.Open(file.Name())
-					defer fileReader.Close()
+					defer func() { _ = fileReader.Close() }()
```

Comment:
I don't think this is required, is it?

Reply:
I got an error...

Reply:
I see why: If `err != nil`, `fileReader` might be `nil`, and the `defer` will panic.
The correct fix is to move the `defer` after the error check (no `func()` required).

Reply:
Hm. You're right, but I'm not sure why we got the error we did (or why my fix worked).

https://github.com/ipfs/go-graphsync/pull/200

Reply:
If that fix worked, it's probably a staticcheck (or was it vet?) bug.